src/SharedTypes.elm

```diff
@@ -0,0 +1,104 @@
+module SharedTypes exposing (..)
+
+import Json.Decode as Decode exposing (Decoder, bool)
+import Json.Decode.Pipeline exposing (optional, required)
```

Comment:
Loved that you decided to use the pipeline package! 💯

Reply:
NIT: I am more of a fan of Json.Extra.Decode than NoRedInk one, but it is definitely better than the `mapX` functions.

Reply:
I've always prefered the pipeline over the Extra.Decode for that reason, mapX always seemed to be a bit off sweat 😅